src/platform/test.rs

```diff
@@ -413,6 +413,31 @@ fn receiver_set() {
     }
 }
 
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn sigterm_handler(_: libc::c_int) -> () {}
+
+#[test]
+#[cfg(not(any(feature = "force-inprocess", target_os = "windows", target_os = "android")))]
+fn receiver_set_eintr() {
+    let (tx0, rx0) = platform::channel().unwrap();
+    let mut rx_set = OsIpcReceiverSet::new().unwrap();
+    rx_set.add(rx0).unwrap();
+    let parent_pid = unsafe {
+        libc::getpid()
+    };
+    unsafe {
+        libc::signal(libc::SIGTERM,
+                     sigterm_handler as fn(libc::c_int) as *mut libc::c_void as libc::sighandler_t);
+    }
+    let _ = unsafe { fork(|| {
+        for _ in 0..10000 {
+            libc::kill(parent_pid, libc::SIGTERM);
+        }
+        tx0.send(b"A", vec![], vec![]).unwrap();
+    })};
+    for _ in rx_set.select().unwrap() {}
```

Comment:
Not sure how useful it is to iterate the results without actually checking them... I'd suggest invoking `next()` to get the first result, like some other `receiver_set()` tests are already doing. (Possibly followed by another `next()` to assert that it's `None`.)